app/src/main/java/com/mashup/mobalmobal/ui/profile/presenter/ProfileViewModel.kt

```diff
@@ -3,52 +3,143 @@ package com.mashup.mobalmobal.ui.profile.presenter
+        BehaviorSubject.createDefault(emptyList())
+
+    init {
+        requestDonations()
             .subscribeOnIO()
-            .doOnSubscribe { _loadingSubject.onNext(true) }
-            .subscribeWithErrorLogger { response ->
-                with(response) {
-                    if (result != Constants.NETWORK_SUCCESS) {
-                        throw IOException(data.message)
-                    }
-                    _profileSubject.onNext(data.toProfileItems())
-                }
-                _loadingSubject.onNext(false)
+            .subscribeWithErrorLogger {
+                _donationsSubject.onNext(it)
+            }
+            .addToDisposables()
+    }
+
+    private val _itemSubject: BehaviorSubject<List<ProfileItem>> = BehaviorSubject.create()
+    val itemSubject: Observable<List<ProfileItem>> = _itemSubject.distinctUntilChanged()
+
+    init {
+        _userDtoSubject.combineLatest(_donationsSubject)
+            .distinctUntilChanged()
+            .map { listOf(it.first.toProfileItem()) + it.second }
+            .subscribeWithErrorLogger {
+                _itemSubject.onNext(it)
             }
             .addToDisposables()
     }
 
+    private fun requestDonations() =
+        requestMyCreateDonations()
+            .zipWith(requestMyDonated())
+            .zipWith(requestMyClosed())
+            .map {
+                val createDonations =
+                    it.first.first.toProfileItems(R.string.profile_header_donation_request)
+                val donatedDoantions =
+                    it.first.second.toProfileItems(R.string.profile_header_donated)
+                val closedDonations =
+                    it.second.toProfileItems(R.string.profile_header_donation_closed)
+
+                mutableListOf<ProfileItem>().apply {
+                    addAll(
+                        Triple(
+                            createDonations.getDonationCount(),
+                            donatedDoantions.getDonationCount(),
+                            closedDonations.getDonationCount()
+                        ).toSummaryProfileItem()
+                    )
+
+                    addAll(createDonations + donatedDoantions + closedDonations)
+                }
+            }
+
+    private fun requestMyCreateDonations() =
+        profileRepository.getMyPosts(STATUS_DONATION_BEFORE)
+            .zipWith(profileRepository.getMyPosts(STATUS_DONATION_INPROGRESS))
+            .map { pair ->
+                val beforeDonation = pair.first.data.posts
+                val inProgressDonation = pair.second.data.posts
+                val list = mutableListOf<PostDto>().also {
+                    if (!beforeDonation.isNullOrEmpty()) it.addAll(beforeDonation)
+                    if (!inProgressDonation.isNullOrEmpty()) it.addAll(inProgressDonation)
+                }
+                MyPostDto(list)
+            }
+
+    private fun requestMyDonated() =
+        profileRepository.getMyDonations()
```

Comment:
```suggestion
    private fun requestMyDonated() = profileRepository.getMyDonations()
```
개인차이긴 한데 이렇게 하면 포매팅이 더 예쁜거 같아요!

Reply:
이건 reformat시에 수정되서 안바꾸는게 나을거 같은데 어떻게 생각하시나요?

Reply:
아아 자동 포매팅 된건가요?
아하 네넹